src/app/server-form/page.tsx

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-console */
+import { getState } from '@/src/utils/form-actions';
+import { cookies } from 'next/headers';
+import { TestForm } from './form';
+import { handleForm } from './actions';
+
```

Comment:
can use import-server to ensure this only ever runs on the server. But by default since it's an async function it'll only run on the server.